infra/src/main/kotlin/fi/fta/geoviite/infra/publication/PublicationService.kt

```diff
@@ -349,75 +349,39 @@ class PublicationService @Autowired constructor(
-            kmPostDao.fetchOnlyDraftVersions(false, trackNumberId)
+        val referenceLineTrackNumberIds = referenceLineService.getMany(DRAFT, requestIds.referenceLines).map { rlId ->
+            rlId.trackNumberId
+        }
+        val revertTrackNumberIds = mutableSetOf<IntId<TrackLayoutTrackNumber>>()
+        val draftOnlyTrackNumberIds = mutableSetOf<IntId<TrackLayoutTrackNumber>>()
+        trackNumberService.getMany(DRAFT, referenceLineTrackNumberIds + requestIds.trackNumbers).forEach { tnId ->
+            if (tnId.getDraftType() != DraftType.OFFICIAL) revertTrackNumberIds.add(tnId.id as IntId)
+            if (tnId.getDraftType() == DraftType.NEW_DRAFT) draftOnlyTrackNumberIds.add(tnId.id as IntId)
+        }
+        val revertLocationTrackIds = requestIds.locationTracks.toSet() + draftOnlyTrackNumberIds.flatMap { tnId ->
+            locationTrackDao.fetchOnlyDraftVersions(includeDeleted = true, tnId)
+        }.map(RowVersion<LocationTrack>::id)
+
+        val revertKmPostIds = requestIds.kmPosts.toSet() + draftOnlyTrackNumberIds.flatMap { tnId ->
+            kmPostDao.fetchOnlyDraftVersions(includeDeleted = true, tnId)
         }.map { kp -> kp.id }
 
-        val switches = publishRequestIds.switches.toSet()
-        val (allLocationTracks, allSwitches) = getRevertRequestLocationTrackAndSwitchDependenciesTransitively(
-            locationTracks, locationTracks, switches, switches
-        )
-        val trackNumbers = (publishRequestIds.trackNumbers + publishRequestIds.referenceLines.mapNotNull { id ->
-            referenceLineService.get(DRAFT, id)
-        }.map { rl -> rl.trackNumberId }.filter(trackNumberService::draftExists)).distinct()
-        val referenceLines = (publishRequestIds.referenceLines + publishRequestIds.trackNumbers.mapNotNull { id ->
-            referenceLineService.getByTrackNumber(DRAFT, id)
-        }.filter { line -> line.draft != null }.map { line -> line.id as IntId }).distinct()
+        val referenceLines = requestIds.referenceLines.toSet() + requestIds.trackNumbers.mapNotNull { tnId ->
+            referenceLineService.getByTrackNumber(DRAFT, tnId)
+        }.filter { line -> line.getDraftType() != DraftType.OFFICIAL }.map { line -> line.id as IntId }
 
         return PublishRequestIds(
-            trackNumbers = trackNumbers,
-            referenceLines = referenceLines,
-            locationTracks = allLocationTracks.toList(),
-            switches = allSwitches.toList(),
-            kmPosts = kmPosts.toList()
+            trackNumbers = revertTrackNumberIds.toList(),
+            referenceLines = referenceLines.toList(),
+            locationTracks = revertLocationTrackIds.toList(),
+            switches = requestIds.switches.distinct(),
+            kmPosts = revertKmPostIds.toList()
         )
     }
 
-    private fun getRevertRequestLocationTrackAndSwitchDependenciesTransitively(
```

Comment:
Tämä poistettu nyt kokonaan: Sallitaan jälleen raiteiden/vaihteiden revert ilman että riippuvuudet lähtee. Jos revertissä poistuu koko olio, samalla poistetaan kaikki viitteet siihen (kuten tapahtuu myös delete-toiminnossa). Palautetaan jos osoittautuu huonoksi vedoksi.

Reply:
Hyvä että näitä aletaan purkaa, sillä näihin liittyikin known issueita joita ei viitsitty korjata härmäisyyden ja korjausten lyhyen elinajan takia.